### ISSUE TITLE
SC2034 shouldn’t trigger when setting nameref variables

### ISSUE THREAD

**Issue by koehn** [koalaman/shellcheck#2919]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2034
- My shellcheck version (`shellcheck --version` or "online"): 0.9.0
- [ x ] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [ x ] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

(requires bash ≥ 4.3)
```sh
#!/bin/bash

# outvar is a nameref variable that allows us to set the value of a variable 
# outside our scope
demo_bug() {
   local x="$1"
   local -n outvar=$2

   # do things with x
   x="foo$x"

   outvar="$x"
}

declare foo

demo_bug "bar" foo

echo "$foo"
```

#### Here's what shellcheck currently says:
```
Line 12:
   outvar="$x"
   ^-- [SC2034](https://www.shellcheck.net/wiki/SC2034) (warning): outvar appears unused. Verify use (or export if used externally).
```

#### Here's what I wanted or expected to see:

No error/warnings.
